pkg/storkctl/clusterpair.go

```diff
@@ -258,6 +266,84 @@ func newCreateClusterPairCommand(cmdFactory Factory, ioStreams genericclioptions
 				util.CheckErr(err)
 				return
 			}
+
+			// create backuplocation in source
+			backupLocation := &storkv1.BackupLocation{
+				ObjectMeta: meta.ObjectMeta{
+					Name:      clusterPairName,
+					Namespace: cmdFactory.GetNamespace(),
+					Annotations: map[string]string{
+						skipResourceAnnotation: "true",
+					},
+				},
+				Location: storkv1.BackupLocationItem{},
+			}
+
+			// create backuplocation secrets in both source and destination
+			credentialData := make(map[string][]byte)
+			switch provider {
+			case "s3":
+				if s3AccessKey == "" {
+					util.CheckErr(getMissingParameterError("s3-access-key", "Access Key missing for S3"))
+					return
+				}
+				if s3SecretKey == "" {
+					util.CheckErr(getMissingParameterError("s3-secret-key", "Secret Key missing for S3"))
+					return
+				}
+				if len(s3EndPoint) == 0 {
+					util.CheckErr(getMissingParameterError("s3-endpoint", "Endpoint missing for S3"))
+					return
+				}
+				if len(s3Region) == 0 {
+					util.CheckErr(getMissingParameterError("s3-region", "Region missing for S3"))
+					return
+				}
+				credentialData["endpoint"] = []byte(s3EndPoint)
+				credentialData["accessKeyID"] = []byte(s3AccessKey)
+				credentialData["secretAccessKey"] = []byte(s3SecretKey)
+				credentialData["region"] = []byte(s3Region)
+				credentialData["disableSSL"] = []byte(strconv.FormatBool(disableSSL))
+				backupLocation.Location.Type = storkv1.BackupLocationS3
+			case "azure":
+				if azureAccountName == "" {
+					util.CheckErr(getMissingParameterError("azure-account-name", "Account Name missing for Azure"))
+					return
+				}
+				if azureAccountKey == "" {
+					util.CheckErr(getMissingParameterError("azure-account-key", "Account Key missing for Azure"))
+					return
+				}
+				credentialData["storageAccountName"] = []byte(azureAccountName)
+				credentialData["storageAccountKey"] = []byte(azureAccountKey)
+				backupLocation.Location.Type = storkv1.BackupLocationAzure
+			case "google":
+				if len(googleProjectID) == 0 {
+					util.CheckErr(getMissingParameterError("google-project-id", "Project ID missing for Google"))
+					return
+				}
+				if len(googleJSONKey) == 0 {
+					util.CheckErr(getMissingParameterError("google-json-key", "Json key file missing for Google"))
+					return
+				}
+				// Read the jsonkey file
+				keyData, err := os.ReadFile(googleJSONKey)
+				if err != nil {
+					util.CheckErr(fmt.Errorf("failed to read json key file: %v", err))
+					return
+				}
+				credentialData["accountKey"] = keyData
+				credentialData["projectID"] = []byte(googleProjectID)
+				backupLocation.Location.Type = storkv1.BackupLocationGoogle
+			default:
+				util.CheckErr(fmt.Errorf("provider needs to be either azure, google, s3"))
+				return
+			}
+
+			credentialData["path"] = []byte(bucket)
+			credentialData["type"] = []byte(provider)
```

Comment:
Add a check if provider is one of s3, azure or google

Reply:
The check was there , modified the message though.